db.go

```diff
@@ -492,16 +489,46 @@ func (db *DB) get(key []byte) (y.ValueStruct, error) {
 	}()
 
 	y.NumGets.Add(1)
-	for i := 0; i < len(tables); i++ {
-		vs := tables[i].Get(key)
+	for _, table := range tables {
+		vs := table.Get(key)
 		y.NumMemtableGets.Add(1)
-		if vs.Meta != 0 || vs.Value != nil {
-			return vs, nil
+		if vs.Valid() {
+			return vs
 		}
 	}
 	return db.lc.get(key)
 }
 
+func (db *DB) multiGet(pairs []keyValuePair) {
+	tables := db.getMemTables() // Lock should be released.
+	defer func() {
+		for _, tbl := range tables {
+			tbl.DecrRef()
+		}
+	}()
+	y.NumGets.Add(int64(len(pairs)))
+	var foundCount int
+	for _, table := range tables {
+		for j := range pairs {
```

Comment:
for _, pair := range pairs

Reply:
This way, we cannot modify the pair in the slice because the `pair` is a value, not a pointer.